src/storage/src/table/streaming_table/state_table.rs

```diff
@@ -359,6 +378,38 @@ impl<S: StateStore> StateTable<S> {
 
 // write
 impl<S: StateStore> StateTable<S> {
+    fn pretty_row_op(row_op: &RowOp, data_types: &[DataType]) -> String {
```

Comment:
How about directly implementing this method on `RowOp` itself?

Reply:
It needs `data_types` to be passed, as a result, the function still looks ugly 🥲